docs/user/azure/install_upi_azurestack.md

```diff
@@ -0,0 +1,591 @@
+open(path, "w").write(yaml.dump(data, default_flow_style=False))'
+```
+
+### Resource Group Name and Infra ID
+
+The OpenShift cluster has been assigned an identifier in the form of `<cluster_name>-<random_string>`. This identifier, called "Infra ID", will be used as
+the base name of most resources that will be created in this example. Export the Infra ID as an environment variable that will be used later in this example:
+
+```sh
+export INFRA_ID=`yq r manifests/cluster-infrastructure-02-config.yml status.infrastructureName`
+```
+
+Also, all resources created in this Azure deployment will exist as part of a [resource group][azure-resource-group]. The resource group name is also
+based on the Infra ID, in the form of `<cluster_name>-<random_string>-rg`. Export the resource group name to an environment variable that will be user later:
+
+```sh
+export RESOURCE_GROUP=`yq r manifests/cluster-infrastructure-02-config.yml status.platformStatus.azure.resourceGroupName`
+```
+
+**Optional:** it's possible to choose any other name for the Infra ID and/or the resource group, but in that case some adjustments in manifests are needed.
+A Python script is provided to help with these adjustments. Export the `INFRA_ID` and the `RESOURCE_GROUP` environment variables with the desired names, copy the
+[`setup-manifests.py`](../../../upi/azure/setup-manifests.py) script locally and invoke it with:
+
+```sh
+python3 setup-manifests.py $RESOURCE_GROUP $INFRA_ID
+```
+
+### Create Cluster Credentials
+
+Azure Stack Hub can only operate in `Manual` credentials mode. At this point,
+follow the [documentation for creating manual credentials][manual-credentials].
+
+Please follow the instructions above, but the result should be similar to this example (note the filenames are arbitrary):
+
+```console
+$ cat manifests/*credentials-secret.yaml
+apiVersion: v1
+kind: Secret
+metadata:
+    name: installer-cloud-credentials
+    namespace: openshift-image-registry
+stringData:
+  azure_subscription_id: <subscription-id>
+  azure_client_id: <client-id>
+  azure_client_secret: <secret>
+  azure_tenant_id: <tenant>
+  azure_resource_prefix: <$INFRA_ID>
+  azure_resourcegroup: <$RESOURCE_GROUP>
+  azure_region: <$REGION>
+apiVersion: v1
+kind: Secret
+metadata:
+    name: cloud-credentials
+    namespace: openshift-ingress-operator
+stringData:
+  azure_subscription_id: <subscription-id>
+  azure_client_id: <client-id>
+  azure_client_secret: <secret>
+  azure_tenant_id: <tenant>
+  azure_resource_prefix: <$INFRA_ID>
+  azure_resourcegroup: <$RESOURCE_GROUP>
+  azure_region: <$REGION>
+apiVersion: v1
+kind: Secret
+metadata:
+  name: azure-cloud-credentials
+  namespace: openshift-machine-api
+stringData:
+  azure_subscription_id: <subscription-id>
+  azure_client_id: <client-id>
+  azure_client_secret: <secret>
+  azure_tenant_id: <tenant>
+  azure_resource_prefix: <$INFRA_ID>
+  azure_resourcegroup: <$RESOURCE_GROUP>
+  azure_region: <$REGION>
```

Comment:
Suggest to also add `Secret` in openshift-controller-manager namespace

```
apiVersion: v1
kind: Secret
metadata:
    name: azure-cloud-credentials
    namespace: openshift-cloud-controller-manager
stringData:
  azure_subscription_id: <subscription-id>
  azure_client_id: <client-id>
  azure_client_secret: <secret>
  azure_tenant_id: <tenant>
  azure_resource_prefix: <$INFRA_ID>
  azure_resourcegroup: <$RESOURCE_GROUP>
  azure_region: <$REGION>
```

Credentials request would there after https://github.com/openshift/cluster-cloud-controller-manager-operator/pull/104 will be merged